src/components/ngRedux.js

```diff
@@ -1,10 +1,41 @@
 import Connector from './connector';
+import invariant from 'invariant';
+import isFunction from '../utils/isFunction';
+import {createStore, applyMiddleware} from 'redux';
 
 export default function ngReduxProvider() {
-  let reduxStore = undefined;
-  this.setReduxStore = store => reduxStore = store;
+  let _reducer = undefined;
+  let _middlewares = [];
+  let _storeEnhancer = undefined;
 
-  this.$get = () => {
-    return Connector(reduxStore);
+  this.createStoreWith = (reducer, middlewares, storeEnhancer) => {
+  	  invariant(
+        isFunction(reducer),
+        'The reducer parameter passed to createStoreWith must be a Function. Instead received %s.',
+        typeof reducer
+      );
+
+      invariant(
+        !storeEnhancer || isFunction(storeEnhancer),
+        'The storeEnhancer parameter passed to createStoreWith must be a Function. Instead received %s.',
+        typeof storeEnhancer
+      );
+
+      _reducer = reducer;
+      _storeEnhancer = storeEnhancer || createStore;
+      _middlewares = middlewares;
+  };
+
+  this.$get = ($injector) => {
+  	let resolvedMiddleware = [];
+  	for(let middleware of _middlewares) {
+  		if(_.isString(middleware)) {
```

Comment:
No problem, happy to help.

Thanks for making this change, redux + ng-redux is looking really promising for a project that I'm just starting.